### PR TITLE
Calculator UI && searchable product name completed

### PR DESCRIPTION
**Pull request by cheedinmar**

### CODE REVIEW

**Review by darmhoo — COMMENTED** [review]
Seen. Nice work

File: src/components/Autocomplete/AutocompleteSearch.vue [hunk 1]
```diff
@@ -0,0 +1,161 @@
+<template>
+    <div>
+        <div class="dropdown">
+            <input
+                type="text"
+                class="form-control w-full px-5 custom-select bg-white rounded-md py-4 font-semibold text-lg"
+                v-model="inputValue"
+                @input="searchEvent"
+            />
+            <div v-show="inputValue && apiLoaded" class="dropdown-list">
```
Comment (harmlessprince) [inline src/components/Autocomplete/AutocompleteSearch.vue:10]: inputValue and apiLoaded is quite confusing at first glance without going through the whole flow.

I suggest something more descriptive.

inputValue => productName as it is in the query param.
apiLoaded => productFetched  or productRetrieved etc

searchEvent can also be renamed to SearchProductByName or searchProduct or searchProductBySomething.....

Comment (cheedinmar) [reply src/components/Autocomplete/AutocompleteSearch.vue:10]: noted

File: src/components/Autocomplete/AutocompleteSearch.vue [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                };
+            }
+            else {
+                query = {
+                    name: this.inputValue
+                };
+            }
+            try {
+                const fetchProduct = await get(this.apiUrl + queryParam(query));
+                this.itemList = fetchProduct.data.data.data;
```
Comment (teebarg) [inline src/components/Autocomplete/AutocompleteSearch.vue:82]: Use Optional Chaining

File: src/components/Autocomplete/AutocompleteSearch.vue [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                this.apiLoaded = true;
+            } catch (err) {
+                this.$displayErrorMessage(err);
+            }
+        },
+        formatInput(item){
+            if(item !== undefined){
+            let result = '';
```
Comment (teebarg) [inline src/components/Autocomplete/AutocompleteSearch.vue:90]: can you lint your code? Use prettier

File: src/components/Autocomplete/AutocompleteSearch.vue [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            this.apiLoaded = false;
+            this.$emit("childToParent", data);
+        },
+        searchEvent() {
+            this.getproduct();
+        },
+
+        async getproduct() {
+            let query = {};
+            if (this.apiUrl === "/api/inventory") {
```
Comment (teebarg) [inline src/components/Autocomplete/AutocompleteSearch.vue:68]: Are you reusing this AutoComplete component?

File: src/views/Calculator.vue [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+import registerBg2 from "@/assets/img/register_bg_2.png"; 
+import AutoComplete from "@/components/Autocomplete/AutocompleteSearch.vue"
+import calculate from "../utilities/calculator"
+import { get, post } from "../utilities/api"
+export default{
+    data(){
+        return{
+            registerBg2,
+              apiUrls: {
+       getProduct: `/api/inventory`,
```
Comment (teebarg) [inline src/views/Calculator.vue:96]: format your code

File: src/views/Calculator.vue [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+    computed:{
+      businessTypeCalc(){
+        return this.businessTypes.map((item)=>{
+          return item;
+        })
+      },
+      // in progress
+      downpaymentCalc(){
+        let  downPaymentArr = [];
+        this.businessTypes.map((bizType)=>{
```
Comment (teebarg) [inline src/views/Calculator.vue:132]: are you returning this?

File: src/views/Calculator.vue [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+    selectedItem(value) {
+      this.selectedProduct = value;
+    },
+        
+
+    async getProduct() {
+      try {
+        const fetchProduct = await get(this.apiUrls.getProduct + this.product);
+        this.products= fetchProduct.data.data.data
+       const found= this.products.find((item)=>{
```
Comment (teebarg) [inline src/views/Calculator.vue:174]: No need to create a new variable since you're just returning

File: src/views/Calculator.vue [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+    methods:{
+    selectedItem(value) {
+      this.selectedProduct = value;
+    },
+        
+
+    async getProduct() {
+      try {
+        const fetchProduct = await get(this.apiUrls.getProduct + this.product);
+        this.products= fetchProduct.data.data.data
```
Comment (teebarg) [inline src/views/Calculator.vue:173]: Use optional Chaining

File: src/views/Calculator.vue [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+        })
+        return found
+      } catch (err) {
+        this.$displayErrorMessage(err);
+      }
+    },
+        async getDownPaymentRates() {
+      try {
+        const fetchDownPaymentRates = await get(this.apiUrls.downPaymentRates);
+        this.downPaymentRates = fetchDownPaymentRates.data.data.data;
```
Comment (teebarg) [inline src/views/Calculator.vue:185]: Use Optional Chaining

File: src/views/Calculator.vue (hunk 6 above)
Comment (cheedinmar) [reply src/views/Calculator.vue:132]: yea. I am returning it

**Review by darmhoo — COMMENTED** [review]
looks good to me

File: src/components/Autocomplete/AutocompleteSearch.vue [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+        },
+        formatInput(item){
+            if(item !== undefined){
+            let result = '';
+            if(this.apiUrl !== "/api/inventory"){
+                result = item.name                
+            }
+            else{               
+                result =  `${item.product_name}   ${item.inventory_sku}`
+                if(item.inventory_status.status === 'Repossessed'){
```
Comment (AltaraNg) [inline src/components/Autocomplete/AutocompleteSearch.vue:96]: Inventory reposessed?

File: src/components/Autocomplete/AutocompleteSearch.vue (hunk 2 above)
Comment (AltaraNg) [reply src/components/Autocomplete/AutocompleteSearch.vue:82]: @cheedinmar why is this not resolved?

File: src/views/Calculator.vue [hunk 11]
```diff
@@ -0,0 +1,487 @@
...
+        });
+      } catch (err) {
+        this.$displayErrorMessage(err);
+      }
+    },
+  },
+};
+</script>
+<style scoped>
+@import url("https://fonts.googleapis.com/css2?family=Nanum+Pen+Script&display=swap");
```
Comment (AltaraNg) [inline src/views/Calculator.vue:475]: why are we importing this google font here?

File: src/views/Calculator.vue [hunk 12]
```diff
@@ -0,0 +1,487 @@
...
+          return result.bizId == bizId && result.percent == percent;
+        }
+      )[0];
+
+    },
+
+    downpaymentCalc() {
+      let downPaymentArr = [];
+      this.businessTypes.forEach((bizType) => {
+        return this.downPaymentRates.forEach((paymentRate) => {
```
Comment (AltaraNg) [inline src/views/Calculator.vue:394]: remove return

File: src/views/Calculator.vue [hunk 13]
```diff
@@ -0,0 +1,487 @@
...
+        color = 'bg-green-300'
+      }else{
+         color ='bg-yellow-300'
+      }
+      return  color 
+    
+    },
+    toggletextColor(){
+      let color = '';
+      if(this.selectedDownpayment == null){
```
Comment (AltaraNg) [inline src/views/Calculator.vue:353]: could use switch statement here `switch (this.selectedDownpayment) {
                case 20:
                    color = 'text-blue-500' 
                    break;
            
                default:
                     color = 'text-gray-500'
                    break;
            }
            return color;`

File: src/views/Calculator.vue [hunk 14]
```diff
@@ -0,0 +1,487 @@
...
+      repaymentDuration: 180,
+      calculation: [],
+      fPayment: "",
+      tpayment: "",
+      pPrice: "",
+      rPayment: "",
+      downPaymentRates: [],
+      eachDownpayment: {},
+      totals: [],
+      test1: true,
```
Comment (AltaraNg) [inline src/views/Calculator.vue:320]: Are we using this?

File: src/views/Calculator.vue [hunk 15]
```diff
@@ -0,0 +1,487 @@
...
+      select_product: false,
+      downPaymentArr: [],
```
Comment (AltaraNg) [inline src/views/Calculator.vue:322]: Are we using this?

File: src/views/Calculator.vue [hunk 16]
```diff
@@ -0,0 +1,487 @@
...
+                  px-3
+                  py-2
+                  flex
+                  justify-evenly
+                  items-center
+                  space-x-20
+                "
+                v-if="downpayments.bizId == b_type.id"
+                :class="
+                  downpayments.percent == 20
```
Comment (AltaraNg) [inline src/views/Calculator.vue:244]: Too much ternary operation

File: src/views/Calculator.vue [hunk 17]
```diff
@@ -0,0 +1,487 @@
...
+                  rounded-full
+                   d_buttons
+                  text-white text-2xl
+                  cursor-pointer
+                  transform  hover:scale-125
+                  bg-gray-500 focus:outline-none
+                "
+                @click="getResultMobile(biztype.id, dpayments.percent)"
+                :class="
+                  dpayments.percent == 20
```
Comment (AltaraNg) [inline src/views/Calculator.vue:81]: Too much ternary

File: src/views/Calculator.vue (hunk 11 above)
Comment (cheedinmar) [reply src/views/Calculator.vue:475]: it is used on the UI for mobile

File: src/views/Calculator.vue (hunk 13 above)
Comment (cheedinmar) [reply src/views/Calculator.vue:353]: ok

File: src/views/Calculator.vue (hunk 14 above)
Comment (cheedinmar) [reply src/views/Calculator.vue:320]: I wil go through all the data property and remove the ones we are not using

File: src/views/Calculator.vue (hunk 16 above)
Comment (cheedinmar) [reply src/views/Calculator.vue:244]: ok I will make it a function

File: src/components/Autocomplete/AutocompleteSearch.vue (hunk 4 above)
Comment (cheedinmar) [reply src/components/Autocomplete/AutocompleteSearch.vue:68]: @darmhoo kindly treat

File: src/components/Autocomplete/AutocompleteSearch.vue (hunk 10 above)
Comment (cheedinmar) [reply src/components/Autocomplete/AutocompleteSearch.vue:96]: @darmhoo

**Review by AltaraNg — APPROVED** [review]

**Pull request merged**